src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooBuildPlanService.java

```diff
@@ -172,10 +171,9 @@ private Stage createBuildStage(ProgrammingLanguage programmingLanguage, ProjectT
          */
         Collection<String> activeProfiles = Arrays.asList(env.getActiveProfiles());
 
-        // Do not run the builds in extra docker containers if the dev-profile is active
-        // Xcode has no dockerfile, it only runs on agents (e.g. sb2-agent-0050562fddde)
-        if (!activeProfiles.contains(JHipsterConstants.SPRING_PROFILE_DEVELOPMENT) && !ProjectType.XCODE.equals(projectType)) {
-            defaultJob.dockerConfiguration(dockerConfigurationImageNameFor(programmingLanguage, Optional.ofNullable(projectType)));
+        // Xcode has no dockerfile, it only runs on macOS agents
+        if (!ProjectType.XCODE.equals(projectType)) {
```

Comment:
I removed the check since our new docker Atlassian config supports docker agents as well. Unfortunately this is not merged yet. I'll remove this change from this PR